lib/galaxy/model/__init__.py

```diff
@@ -289,11 +289,13 @@ def metrics(self):
 
     def set_streams(self, tool_stdout, tool_stderr, job_stdout=None, job_stderr=None, job_messages=None):
         def shrink_and_unicodify(what, stream):
-            stream = galaxy.util.unicodify(stream) or u''
-            if (len(stream) > galaxy.util.DATABASE_MAX_STRING_SIZE):
-                stream = galaxy.util.shrink_string_by_size(tool_stdout, galaxy.util.DATABASE_MAX_STRING_SIZE, join_by="\n..\n", left_larger=True, beginning_on_size_error=True)
```

Comment:
Also this was a bug imo, it should have been `stream`, not `tool_stdout`. So for large stderr streams we'd get the stdout again ...